### PR TITLE
feat(ec2): create Peers via security group ids

### PR DESCRIPTION
**Pull request by jacobklitzke**
Allows users to add ingress/egress security group rules containing a security group id using the Peer interface.

Implements #7111

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/lib/peer.ts [hunk 1]
```diff
@@ -75,6 +75,13 @@ export class Peer {
     return new PrefixList(prefixListId);
   }
 
+  /**
+   * A security group ID
+   */
+  public static securityGroupId(securityGroupId: string, sourceSecurityGroupOwnerId?: string): IPeer {
```
Comment (njlynch) [inline packages/@aws-cdk/aws-ec2/lib/peer.ts:81]: I think there's an argument that the `sourceSecurityGroupOwnerId` parameter should be mandatory. If the parameter is omitted, this generates an identical rule config to a SecurityGroup itself, which doesn't really provide (much) value.

Do you see a compelling reason to keep it optional?

**Review by njlynch — CHANGES_REQUESTED** [review]
Looks good!

However, I do question whether allowing the ownership information to be optional makes sense, given it then just replicates the behavior of `SecurityGroup`.

Also, can you update the docstring for the `Peer` class as part of this? The last paragraph explicitly mentions Security Groups and looks like it needs a slight update as part of this.

File: packages/@aws-cdk/aws-ec2/lib/peer.ts (hunk 1 above)
Comment (jacobklitzke) [reply packages/@aws-cdk/aws-ec2/lib/peer.ts:81]: The only reason I can think to keep `sourceSecurityGroupOwnerId` optional is to mimic the behavior of the `AWS::EC2::SecurityGroup` resource type in [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-rule-1.html#cfn-ec2-security-group-rule-sourcesecuritygroupownerid). Users are not required to put a `SourceSecurityGroupOwnerId` if the security group is in the same account. I agree that this duplicates existing functionality in CDK, but it does provide additional flexibility for how a rule containing a security group id can be added to a new security group in CDK. 

Requiring the `sourceSecurityGroupOwnerId` parameter will not cause any issues if a user specifies a security group which exists in the same account as the new `SecurityGroup`. Requiring the parameter will just slightly deviate from the behavior outlined in CloudFormation.

Comment (jacobklitzke) [reply packages/@aws-cdk/aws-ec2/lib/peer.ts:81]: @njlynch What do you think of the above reasoning?

**Review by njlynch — APPROVED** [review]

File: packages/@aws-cdk/aws-ec2/lib/peer.ts (hunk 1 above)
Comment (njlynch) [reply packages/@aws-cdk/aws-ec2/lib/peer.ts:81]: Fair enough. It's a bit duplicative, but not harming anything. Thanks.

**Pull request merged**